ReactSkia/RSkPlatformPlugin/src/interface/appstate/AppStateBase.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include "folly/Format.h"
+#include "folly/io/async/ScopedEventBaseThread.h"
+#include "ReactSkia/RSkPlatformPlugin/src/RnsPlugin.h"
+
+namespace facebook {
+namespace react {
+namespace rnsplugin {
+
+using namespace boost;
+using namespace folly;
+using namespace std;
+using folly::format;
+using std::string;
+using std::vector;
+
+class AppStateBase {
+ public:
+	class Client {
+	 public:
+		virtual ~Client() {}
+		// Events
+		virtual void onChange(AppStateBase &base) = 0;
+		virtual void onMemoryWarning(AppStateBase &base) = 0;
+		virtual void onFocus(AppStateBase &base) = 0;
+		virtual void onBlur(AppStateBase &base) = 0;
+	};
+
+	AppStateBase(AppStateBase::Client &client)
+	: appStateName_(format("RNSAppState-{}",nextUniqueId()).str()),
+		/* appStateThread_(!appStateName_.empty()? appStateName_.c_str(): "RNSAppState-UnknownID"), */
+		isAppStateBaseValid(true),
+		client_(client) {}
+	
+	virtual ~AppStateBase(){
+		isAppStateBaseValid = false;
+	}
+
+	std::string currentState() { return currentState_; }
+	virtual void setCurrentState(std::string currentState) = 0;
+
+ private:
+	static uint64_t nextUniqueId();
+	std::string appStateName_;
+	/* ScopedEventBaseThread appStateThread_; */
+	bool isAppStateBaseValid{false};
+	Client &client_;
+	std::string currentState_{"active"};
+
+ protected:
+	void dispatchOnChange() {
+		client_.onChange(*this);
+		/*appStateThread_.getEventBase()->runInEventBaseThread([&] {
+		if (isAppStateBaseValid)
```

Comment:
Keep macro,
Async_Event_Handeling = false
//TODO: disabling temporarily now due to crash issue seeing in folly thread usage.